File: php/src/BasicItem.php
<?php

declare(strict_types=1);

namespace App;

class BasicItem implements Item
{
    private string $name;
    private int $baseDamage;
    private float $damageModifier;

    public function __construct(
        string $name,
        int $baseDamage,
        float $damageModifier
    ) {
        $this->name = $name;
        $this->baseDamage = $baseDamage;
        $this->damageModifier = $damageModifier;
    }

    public function getBaseDamage(): int
    {
        return $this->baseDamage;
    }

    public function getDamageModifier(): float
    {
        return $this->damageModifier;
    }
}

File: go/basic_item.go
package codingdojo

type BasicItem struct {
	name           string
	baseDamage     int32
	damageModifier float64
}

func MakeBasicItem(name string, baseDamage int32, damageModifier float64) BasicItem {
	return BasicItem{name, baseDamage, damageModifier}
}

func (i BasicItem) GetBaseDamage() int32 {
	return i.baseDamage
}

func (i BasicItem) GetDamageModifier() float64 {
	return i.damageModifier
}
